sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -431,7 +436,7 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
           val hdfsPath = new Path(path)
           val fs = hdfsPath.getFileSystem(hadoopConf)
           val qualified = hdfsPath.makeQualified(fs.getUri, fs.getWorkingDirectory)
-
+          inputExists = inputExists && fs.exists(qualified)
```

Comment:
This can be quite expensive since each `fs.exists(qualified)` call invokes `FileSystem.getFileStatus()`, which is an RPC call. On the other hand, we've already called `fs.listStatus(qualified)` below. Would be better to merge these two.